packages/contracts/contracts/factory/DAOFactory.sol

```diff
@@ -2,155 +2,152 @@
+        // Register DAO.
+        daoRegistry.register(dao, msg.sender, _daoSettings.name);
 
-        setDAOPermissions(dao, address(voting));
-
-        emit DAOCreated(_daoConfig.name, address(token), address(voting));
-    }
+        // Grant the temporary permissions.
+        // Grant Temporarly `ROOT_PERMISSION_ID` to `pluginSetupProcessor`.
+        dao.grant(address(dao), address(pluginSetupProcessor), dao.ROOT_PERMISSION_ID());
 
-    /// @notice Creates a new DAO with the `AllowlistVoting` component installed.
-    /// @param _daoConfig The name and metadata hash of the DAO.
-    /// @param _voteConfig The configuration used to set up the the majority voting.
-    /// @param _allowlistVoters An array of addresses that are allowed to vote.
-    /// @param _trustedForwarder The address of the trusted forwarder required for meta transactions.
-    function createAllowlistVotingDAO(
-        DAOConfig calldata _daoConfig,
-        VoteConfig calldata _voteConfig,
-        address[] calldata _allowlistVoters,
-        address _trustedForwarder
-    ) external returns (DAO dao, AllowlistVoting voting) {
-        dao = createDAO(_daoConfig, _trustedForwarder);
-
-        daoRegistry.register(dao, msg.sender, _daoConfig.name);
-
-        voting = createAllowlistVoting(dao, _allowlistVoters, _voteConfig);
-
-        setDAOPermissions(dao, address(voting));
+        // Grant Temporarly `PROCESS_INSTALL_PERMISSION_ID` on `pluginSetupProcessor` to this `DAOFactory`.
+        dao.grant(
+            address(pluginSetupProcessor),
+            address(this),
+            pluginSetupProcessor.PROCESS_INSTALL_PERMISSION_ID()
+        );
+
+        // Install plugins on the newly created DAO.
+        bool hasGovernancePlugin;
+
+        for (uint256 i = 0; i < pluginSettings.length; i++) {
+            // Prepare plugin.
+            (
+                address plugin,
+                ,
+                BulkPermissionsLib.ItemMultiTarget[] memory permissions
+            ) = pluginSetupProcessor.prepareInstallation(
+                    address(dao),
+                    pluginSettings[i].pluginSetup,
+                    pluginSettings[i].pluginSetupRepo,
+                    pluginSettings[i].data
+                );
+
+            // Check if plugin is a governance plugin.
+            if (!hasGovernancePlugin) {
+                hasGovernancePlugin = hasExecutePermission(permissions);
+            }
+
+            // Apply plugin.
+            pluginSetupProcessor.applyInstallation(
+                address(dao),
+                pluginSettings[i].pluginSetup,
+                plugin,
+                permissions
+            );
+        }
+
+        // Check one of the plugin was governance plugin.
+        if (!hasGovernancePlugin) {
+            revert NoGovernancePluginProvided();
+        }
+
+        // Set the rest of DAO's permissions TODO: only set, so it is extendable.
+        _setDAOPermissions(dao);
```

Comment:
Should we add support for sub-Dao permission in this iteration?
`_setDAOPermissions(dao, subDaoPermission);`
if we add this feature, DAOs can create subDAOs with limited/extended permissions instead of the default ones set now by `_setDAOPermissions(dao)`